diff --git a/daemon.vala b/daemon.vala
index d09564c..a1ad05f 100644
--- a/daemon.vala
+++ b/daemon.vala
@@ -117,9 +117,12 @@ internal class AstalNotifd.Daemon : Object {
         int expire_timeout
     ) throws DBusError, IOError {
         if (hints.get("image-data") != null) {
-            var file = cache_image(hints.get("image-data"), app_name);
+            var image = hints.get("image-data");
+            var file = cache_image(image, app_name);
             if (file != null) {
                 hints.set("image-path", new Variant.string(file));
+                hints.set("image-width", image.get_child_value(0));
+                hints.set("image-height", image.get_child_value(1));
                 hints.remove("image-data");
             }
         }
@@ -204,7 +207,7 @@ internal class AstalNotifd.Daemon : Object {
     }
 
     public string[] get_capabilities() throws DBusError, IOError {
-        return {"action-icons", "actions", "body", "icon-static", "persistence", "sound"};
+        return {"action-icons", "actions", "body", "body-markup", "icon-static", "persistence", "sound", "x-canonical-append"};
     }
 
     private string? cache_image(Variant image, string app_name) {
diff --git a/notification.vala b/notification.vala
index c28138f..8abe2f0 100644
--- a/notification.vala
+++ b/notification.vala
@@ -46,6 +46,8 @@ public class AstalNotifd.Notification : Object {
 
     /** Path of an image  */
     public string image { get { return get_str_hint("image-path"); } }
+    public int image_width { get { return get_int_hint("image-width"); } }
+    public int image_height { get { return get_int_hint("image-height"); } }
 
     /**
      * Indicates whether [struct@AstalNotifd.Action]
@@ -61,6 +63,9 @@ public class AstalNotifd.Notification : Object {
     /** Specifies the name of the desktop filename representing the calling program. */
     public string desktop_entry { get { return get_str_hint("desktop-entry"); } }
 
+    /** Indicates whether notification should be collapsed.  */
+    public bool append { get { return get_bool_hint("x-canonical-append"); } }
+
     /** Indicates whether notification is kept after action invocation.  */
     public bool resident { get { return get_bool_hint("resident"); } }
 
